internal/registration/registration_test.go

```diff
@@ -0,0 +1,172 @@
+package registration_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	osUtil "os"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	pb "github.com/redhatinsights/yggdrasil/protocol"
+
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-device-worker/internal/datatransfer"
+	"github.com/jakub-dzon/k4e-device-worker/internal/hardware"
+	"github.com/jakub-dzon/k4e-device-worker/internal/heartbeat"
+	"github.com/jakub-dzon/k4e-device-worker/internal/os"
+	"github.com/jakub-dzon/k4e-device-worker/internal/registration"
+	"github.com/jakub-dzon/k4e-device-worker/internal/workload"
+)
+
+const (
+	deviceConfigName = "device-config.json"
+)
+
+var _ = Describe("Registration", func() {
+
+	var (
+		datadir        string
+		mockCtrl       *gomock.Controller
+		wkManager      *workload.WorkloadManager
+		wkwMock        *workload.MockWorkloadWrapper
+		dispatcherMock *registration.MockDispatcherClient
+		configManager  *configuration.Manager
+		hb             *heartbeat.Heartbeat
+		hw             = &hardware.Hardware{}
+		monitor        = &datatransfer.Monitor{}
+		os             = &os.OS{}
+		err            error
+	)
+
+	BeforeEach(func() {
+		datadir, err = ioutil.TempDir("", "registrationTest")
+		Expect(err).ToNot(HaveOccurred())
+
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		wkwMock = workload.NewMockWorkloadWrapper(mockCtrl)
+		wkwMock.EXPECT().Init().Return(nil).AnyTimes()
+
+		dispatcherMock = registration.NewMockDispatcherClient(mockCtrl)
+
+		wkManager, err = workload.NewWorkloadManagerWithParams(datadir, wkwMock)
+		Expect(err).NotTo(HaveOccurred(), "Cannot start the Workload Manager")
+
+		configManager = configuration.NewConfigurationManager(datadir)
+
+		hb = heartbeat.NewHeartbeatService(dispatcherMock,
+			configManager,
+			wkManager,
+			hw,
+			monitor)
+
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+		_ = osUtil.Remove(datadir)
+	})
+
+	Context("RegisterDevice", func() {
+
+		RegistrationMatcher := func() gomock.Matcher {
+			return regMatcher{}
+		}
+
+		It("Work as expected", func() {
+
+			// given
+			reg := registration.NewRegistration(
+				hw, os, dispatcherMock, configManager, hb, wkManager, monitor)
+
+			// then
+			dispatcherMock.EXPECT().Send(gomock.Any(), RegistrationMatcher()).Times(1)
```

Comment:
What do you think about validating data being send?

Reply:
I understand we are testing this [line](https://github.com/jakub-dzon/k4e-device-worker/blob/main/internal/registration/registration.go#L94). My question was whether it makes sense to verify data attribute.